### PR TITLE
Merge common v4.0.1 release to dev

### PR DESCRIPTION
**Pull request by iamgusain**
Merging release/4.0.0 branch back to dev

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java [hunk 1]
```diff
@@ -715,6 +715,29 @@ private synchronized KeyPair generateKeyPairFromAndroidKeyStore()
             final Locale currentLocale = Locale.getDefault();
             applyKeyStoreLocaleWorkarounds(currentLocale);
 
+            /*
```
Comment (rpdome) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java:718]: Hm. This does not seem to be ported to common4j (AndroidWrappedKeyLoader, line 249)

StorageHelper is already marked as deprecated and won't be used in 4j.

File: common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java [hunk 2]
```diff
@@ -135,6 +137,15 @@ public static String join(final char delimiter, @NonNull final Iterable<String>
                 && !StringExtensions.isNullOrBlank(uidUtidArray[INDEX_UTID])) {
             uid = uidUtidArray[INDEX_UID];
             utid = uidUtidArray[INDEX_UTID];
+        } else {
```
Comment (rpdome) [inline common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java:140]: This method has been migrated to 4j (same class name), and this change should go there as well.

Comment (iamgusain) [reply common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java:140]: added

File: common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java (hunk 1 above)
Comment (iamgusain) [reply common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java:718]: Lets do that in a separate change

**Review by shoatman — APPROVED** [review]

**Review by rpdome — APPROVED** [review]

**Pull request merged**